crates/re_types/src/loggable_batch.rs

```diff
@@ -7,63 +7,112 @@ use crate::Archetype;
     // type Loggable: Loggable;
 
-    /// The fully-qualified name of this list, e.g. `rerun.datatypes.Vec2D`.
+    /// The fully-qualified name of this batch, e.g. `rerun.datatypes.Vec2D`.
     fn name(&self) -> Self::Name;
 
-    /// The number of component instances stored into this list.
+    /// The number of component instances stored into this batch.
     fn num_instances(&self) -> usize;
 
     /// The underlying [`arrow2::datatypes::Field`], including datatype extensions.
     fn arrow_field(&self) -> arrow2::datatypes::Field;
 
-    /// Serializes the list into an Arrow array.
+    /// Serializes the batch into an Arrow array.
     ///
-    /// This will _never_ fail for Rerun's built-in [`LoggableList`].
-    /// For the non-fallible version, see [`LoggableList::to_arrow`].
+    /// This will _never_ fail for Rerun's built-in [`LoggableBatch`].
+    /// For the non-fallible version, see [`LoggableBatch::to_arrow`].
     fn try_to_arrow(&self) -> SerializationResult<Box<dyn ::arrow2::array::Array>>;
 
-    /// Serializes the list into an Arrow array.
+    /// Serializes the batch into an Arrow array.
     ///
     /// Panics on failure.
-    /// This will _never_ fail for Rerun's built-in [`LoggableList`]s.
+    /// This will _never_ fail for Rerun's built-in [`LoggableBatch`]s.
     ///
-    /// For the fallible version, see [`LoggableList::try_to_arrow`].
+    /// For the fallible version, see [`LoggableBatch::try_to_arrow`].
     fn to_arrow(&self) -> Box<dyn ::arrow2::array::Array> {
         self.try_to_arrow().detailed_unwrap()
     }
 }
 
-/// A [`DatatypeList`] represents an array's worth of [`Datatype`] instances.
+/// A [`DatatypeBatch`] represents an array's worth of [`Datatype`] instances.
 ///
-/// Any [`LoggableList`] with a [`Loggable::Name`] set to [`DatatypeName`] automatically
-/// implements [`DatatypeList`].
-pub trait DatatypeList: LoggableList<Name = DatatypeName> {}
+/// Any [`LoggableBatch`] with a [`Loggable::Name`] set to [`DatatypeName`] automatically
+/// implements [`DatatypeBatch`].
+pub trait DatatypeBatch: LoggableBatch<Name = DatatypeName> {}
 
-/// A [`ComponentList`] represents an array's worth of [`Component`] instances.
+/// A [`ComponentBatch`] represents an array's worth of [`Component`] instances.
 ///
-/// Any [`LoggableList`] with a [`Loggable::Name`] set to [`ComponentName`] automatically
-/// implements [`ComponentList`].
-pub trait ComponentList: LoggableList<Name = ComponentName> {}
+/// Any [`LoggableBatch`] with a [`Loggable::Name`] set to [`ComponentName`] automatically
+/// implements [`ComponentBatch`].
+pub trait ComponentBatch: LoggableBatch<Name = ComponentName> {}
+
+/// Holds either an owned [`ComponentBatch`] that lives on heap, or a reference to one.
+///
+/// This doesn't use [`std::borrow::Cow`] on purpose: `Cow` requires `Clone`, which would break
+/// object-safety, which would prevent us from erasing [`ComponentBatch`]s in the first place.
+pub enum AnyComponentBatch<'a> {
```

Comment:
is "any" really the right name for this concept? `MaybeOwned` is too clunky admittedly :/